src/Microsoft.AspNetCore.SignalR.Client.Core/IConnectionFactory.cs

```diff
@@ -1,15 +1,16 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Connections;
 
 namespace Microsoft.AspNetCore.SignalR.Client
 {
     public interface IConnectionFactory
     {
-        Task<ConnectionContext> ConnectAsync(TransferFormat transferFormat);
+        Task<ConnectionContext> ConnectAsync(TransferFormat transferFormat, CancellationToken cancellationToken);
 
         Task DisposeAsync(ConnectionContext connection);
```

Comment:
What about DisposeAsync?

Reply:
> DisposeAsync accepting a CancellationToken: while in theory it makes sense that anything async can be canceled, disposal is about cleanup, closing things out, free'ing resources, etc., which is generally not something that should be canceled; cleanup is still important for work that's canceled. The same CancellationToken that caused the actual work to be canceled would typically be the same token passed to DisposeAsync, making DisposeAsync worthless because cancellation of the work would cause DisposeAsync to be a nop. If someone wants to avoid being blocked waiting for disposal, they can avoid waiting on the resulting Task, or wait on it only for some period of time.

https://github.com/dotnet/csharplang/blob/195efa07806284d7b57550e7447dc8bd39c156bf/proposals/async-streams.md#iasyncdisposable

Reply:
Should we remove it from StopAsync?

Reply:
Stop isn't quite the same as Dispose though. Dispose is about releasing resources. Stop is about graceful shutdown of the connection. I think cancelling stop would make sense, but it should put the connection in the "disgracefully terminated" state, rather than restoring it back to running. It can cancel the DELETE request, and other shutdown processes but it should continue to shut down all the local resources associated with the connection (by Disposing the HttpConnection, which isn't cancelable as per above).

Today, we don't do anything with the CT in StopAsync though, so it's somewhat moot whether we keep it or not. If we remove it, adding it back via default parameter is breaking but via overload wouldn't be breaking.